File: Java/FizzBuzz.java
public class FizzBuzz {
    
    public static void main(String[] args) {
       fizzBuzzClear();
       fizzBuzzDRY();
       fizzBuzzDRY_2();
    }

    public static void fizzBuzzClear(){
        for (int i = 1; i <= 100; i++) {
            if (i % 15 == 0) {
                System.out.println("FizzBuzz");
            } else if (i % 3 == 0) {
                System.out.println("Fizz");
            } else if (i % 5 == 0) {
                System.out.println("Buzz");
            } else {
                System.out.println(String.valueOf(i));
            }
        }
    }

    public static void fizzBuzzDRY(){
        final String EMPTY = "";
        for (int i = 1; i <= 100; i++) {
            String value = EMPTY;
            if (i % 3 == 0) value += "Fizz";
            if (i % 5 == 0) value += "Buzz";
            if (value == EMPTY) value += i;
            System.out.println(value);
        }
    }

    public static void fizzBuzzDRY_2(){
        StringBuilder builder = new StringBuilder(1000);
        for (int i = 1; i <= 100; i++) {
            final int length = builder.length();
            if (i % 3 == 0) builder.append("Fizz");
            if (i % 5 == 0) builder.append("Buzz");
            if (length == builder.length()) builder.append(i);
            builder.append('\n');
        }
        System.out.println(builder);
    }


}

File: Ruby/fizzbuzz.rb
puts fizzbuzz


def fizzbuzz
  res=""
  (1..100). to_a.each do |elem|
    res+="Fizz" if (elem%3).zero?
    res+="Buzz" if (elem%5).zero?
    res+="#{elem}" unless ((elem%3).zero? or (elem%5).zero?);
    res+=","
  end
  res
end

def fizzbuzz_2
  (1..100).each do |i|
    puts (a = [
            (i % 3).zero? ? "Fizz" : nil,
            (i % 5).zero? ? "Buzz" : nil
    ].compact.join).empty? ? i : a
  end
end


def fizzbuzz_3
  (1..100).map do |i|
    puts i, [
            (i % 3).zero? ? "Fizz" : nil,
            (i % 5).zero? ? "Buzz" : nil
    ].compact.join
  end.compact
end

def fizzbuzz_4
  100.times do |i|
    num=i+1
    [ (num % 3).zero? ? "Fizz" : nil,
      (num % 5).zero? ? "Buzz" : nil,
      ((num % 3).zero? or (num % 5).zero?) ? nil : num,
      ","
      ].compact.join.display
  end
end

def fizzbuzz_5
  (1..100).each do |i|
    puts (s = "#{"Fizz" if (i % 3).zero?}#{"Buzz" if (i % 5).zero?}").empty? ? i : s
  end
end
